rsyncd/rsyncd_test.go

```diff
@@ -0,0 +1,37 @@
+package rsyncd_test
+
+import (
+	"context"
+	"log"
+	"net"
+	"time"
+
+	"github.com/gokrazy/rsync/rsyncd"
+)
+
+func ExampleNewServer() {
+	// simulate user (or process supervisor) asking us to stop soon after starting
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
+
+	listener, err := net.Listen("tcp", "127.0.0.1:873")
```

Comment:
nit: use `localhost:873` here, no need to make this IPv4-specific (running this code would fail on an IPv6-only machine)

Reply:
Oh wow, nice catch! Definitely needs to be fixed. Learned new once again :)